src/Design/03_InitialPoC_trans.purs

```diff
@@ -1,56 +1,52 @@
 module PSPS.Design.InitialPoC.Trans where
 
-import PSPS (class Bind, class Eval, App, kind Expr, kind ModuleName)
+import FFI.PSPS.Design.InitialPoC.Trans (class FFI_PSPS_Design_InitialPoC_Trans_Eval, Foreign'intAdd, Foreign'intMul)
+import PSPS (class Bind, class Eval, Abs, App, Foreign, kind Expr, kind ModuleName)
+import Type.Data.Peano (Succ, Z)
 
 foreign import data PSPS_Design_InitialPoC_Trans :: ModuleName
 
 class PSPS_Design_InitialPoC_Trans_Eval (expr :: Expr) (result :: Expr) | expr -> result
 
 class PSPS_Design_InitialPoC_Trans_Bind (path :: Symbol) (expr :: Expr) | path -> expr
 
-foreign import kind Maybe
-
-foreign import data Just :: Expr -> Maybe
-
-foreign import data Nothing :: Maybe
-
-foreign import data Maybe0 :: Maybe -> Expr
-
-foreign import data Maybe1 :: (Expr -> Maybe) -> Expr
-
-instance iPSPS_Design_InitialPoC_Trans_Bind'Just :: PSPS_Design_InitialPoC_Trans_Bind "Just" (Maybe1 Just)
-
-instance iPSPS_Design_InitialPoC_Trans_Bind'Nothing :: PSPS_Design_InitialPoC_Trans_Bind "Nothing" (Maybe0 Nothing)
-
-instance iBind'Just :: PSPS_Design_InitialPoC_Trans_Bind "Just" o => Bind PSPS_Design_InitialPoC_Trans "Just" o
+instance iPSPS_Design_InitialPoC_Trans_Bind'intAdd :: PSPS_Design_InitialPoC_Trans_Bind "intAdd" (Abs "intAdd" (Succ Z) (Abs "intAdd" Z (Foreign Foreign'intAdd)))
```

Comment:
In general, abstraction works the same way that it does in the lambda calculus.  In the lambda calculus, we'd have `(\x.(\y.z))`. Here, we have two levels of abstraction, followed by a reference to the FFI (`Foreign Foreign'intAdd`).  Remember that `Bind` is building a lookup table, so when I ask for `"intAdd", I'll get back `(Abs "intAdd" (Succ Z) (Abs "intAdd" Z (Foreign Foreign'intAdd)))`. This is the same structure we'll use for `Eval` later on.